### PR TITLE
[#5128] Implement LiquibaseCustomizer for SpringLiquibase

### LINKED ISSUE #5128: LiquibaseCustomizer in SpringLiquibase to register a ChangeExecListener in Spring Boot
### Search first

- [X] I searched and no similar issues were found

### Description

I am using Spring Boot and would like to register a ChangeExecListener. 

This is not possible without:
- Inherit SpringLiquibase and override createLiquibase.
- Copy the LiquibaseAutoConfiguration into our codebase and return the inherited SpringLiquibase in LiquibaseAutoConfiguration.LiquibaseConfiguration#createSpringLiquibase.

A LiquibaseCustomizer interface could be applied in createLiquibase, giving the required flexibility to Spring Boot. 

See also https://github.com/spring-projects/spring-boot/issues/38055

### Steps To Reproduce

N/A

### Expected/Desired Behavior

An extension point like a LiquibaseCustomizer interface is available in SpringLiquibase to register an ChangeExecListener

### Liquibase Version

4.22.0

### Database Vendor & Version

_No response_

### Liquibase Integration

Spring Boot

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

_No response_

### Additional Context

_No response_

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR (Thank you!)

### PR DESCRIPTION
**Pull request by jwgmeligmeyling**
Fixes #5128

## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [x] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: liquibase-standard/src/main/java/liquibase/integration/spring/SpringLiquibase.java [hunk 1]
```diff
@@ -366,6 +368,10 @@ protected Liquibase createLiquibase(Connection c) throws LiquibaseException {
             ((DerbyDatabase) liquibase.getDatabase()).setShutdownEmbeddedDerby(false);
         }
 
+        if (customizer != null) {
+            customizer.customize(liquibase);
+        }
+
 		return liquibase;
```
Comment (jwgmeligmeyling) [inline liquibase-standard/src/main/java/liquibase/integration/spring/SpringLiquibase.java:375]: The file already used a mix of tab and space based indentation (if statement is aligned with the closing bracket from the statement above)

Comment (jwgmeligmeyling) [reply liquibase-standard/src/main/java/liquibase/integration/spring/SpringLiquibase.java:375]: I’m an external contributor too, lets see what the maintainers have to say :)

**Comment by jwgmeligmeyling** [comment]
I’m an external contributor too, lets see what the maintainers have to say :)

**Comment by filipelautert** [comment]
Hello @asashour @jwgmeligmeyling ! Thanks for both PRs! 
The parameterized customizer that we have here seems more flexible, but #5659  has tests. Maybe @jwgmeligmeyling could rebase the code and reproduce @asashour tests here so we can move ahead with this PR.

**Comment by filipelautert** [comment]
Implemented under PR #5741 . Thanks!

**Pull request closed without merging**